pennylane/decomposition/decomposition_rule.py

```diff
@@ -0,0 +1,337 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Defines the ``DecompositionRule`` class to represent a decomposition rule."""
+
+from __future__ import annotations
+
+import inspect
+from collections import defaultdict
+from textwrap import dedent
+from typing import Callable, Optional, Type, overload
+
+from pennylane.operation import Operator
+
+from .resources import CompressedResourceOp, Resources, resource_rep
+
+
+@overload
+def register_resources(resources: Callable | dict) -> Callable[[Callable], DecompositionRule]: ...
+@overload
+def register_resources(resources: Callable | dict, qfunc: Callable) -> DecompositionRule: ...
+def register_resources(resources: Callable | dict, qfunc: Optional[Callable] = None):
+    """Bind a quantum function to its required resources.
+
+    .. note::
+
+        This function is only relevant when the new experimental graph-based decomposition system
+        (introduced in v0.41) is enabled via ``qml.decompositions.enable_graph()``. This new way of
+        doing decompositions is generally more performant and accommodates multiple alternative
```

Comment:
```suggestion
        doing decompositions is generally more resource efficient and accommodates multiple alternative
```
When I see "performant", I would think it runs faster. Though I havne't run the numbers, I would be skeptical if we had less classical overhead in this ecosystem. 

Similar comment on the other uses.